src/main/java/com/eventsphere/user/service/UserService.java

```diff
@@ -0,0 +1,225 @@
+     * @return The {@link User} object.
+     * @throws UserNotFoundException if the user with the given ID is not found.
+     */
+    public User get(Long id) {
+        return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+    }
+
+    /**
+     * Saves a user.
+     *
+     * @param user The {@link User} object to save.
+     * @return The saved {@link User} object.
+     * @throws UserNotValidException if the user data is invalid.
+     */
+    public User save(User user) {
+        User savedUser;
+
+        try {
+            user.setCreatedAt(null);
+            user.setUpdatedAt(null);
+
+            savedUser = userRepository.save(user);
+        } catch (RuntimeException ex) {
+            ex.printStackTrace();
+            throw new UserNotValidException("Invalid User data: " + ex.getMessage());
+        }
+
+        return savedUser;
+    }
+
+    /**
+     * Creates a new user.
+     *
+     * @param user The {@link User} object to create.
+     * @return The created {@link User} object.
+     * @throws UserAlreadyExistsException if a user with the same username or email already exists.
+     */
+    public User create(User user) {
+        User createdUser;
+
+        if (userRepository.existsByUsername(user.getUsername())) {
+            throw new UserAlreadyExistsException("This username is already registered");
+        } else if (userRepository.existsByEmail(user.getEmail())) {
+            throw new UserAlreadyExistsException("This email is already registered");
+        } else {
+            createdUser = save(user);
+        }
+
+        return createdUser;
+    }
+
+    /**
+     * Updates an existing user. (PUT)
+     *
+     * @param user The updated {@link User} object.
+     * @return The updated {@link User} object.
+     * @throws UserNotFoundException      if the user with the given ID is not found.
+     * @throws UserAlreadyExistsException if a user with the updated username or email already exists.
+     */
+    public User update(User user) {
+        User updatedUser = null;
+        User userFromDb = userRepository.findById(user.getId())
+                .orElseThrow(() -> new UserNotFoundException(user.getId()));
+
+        if (checkUsernameUpdate(userFromDb.getUsername(), user.getUsername()) &&
+                checkEmailUpdate(userFromDb.getEmail(), user.getEmail())) {
+            updatedUser = save(user);
+        }
+
+        return updatedUser;
+    }
+
+    /**
+     * Updates an existing user with partial data. (PATCH)
+     *
+     * @param userId  The ID of the user to update.
+     * @param userDto The {@link UserDto} object containing the partial user data.
+     * @return The updated {@link User} object.
+     * @throws UserNotFoundException      if the user with the given ID is not found.
+     * @throws UserAlreadyExistsException if a user with the updated username or email already exists.
+     */
+    public User update(Long userId, UserDto userDto) {
```

Comment:
Please rethink this bit of logic because it will require to add a new if every time if we want to add a new field to the user